src/versions.rs

```diff
@@ -0,0 +1,254 @@
+                let version = match split.get(1) {
+                    Some(r) => r.to_string(),
+                    None => split[0].to_string()
+                };
+                let final_version: String = version.split('-').next().unwrap().to_string();
+
+                return Some(final_version);
+            }
+        }
+    }
+
+    None
+}
+fn find_xbps_package(name: &str) -> Option<String> {
+    let file: File = match File::open("/var/db/xbps/pkgdb-0.38.plist") {
+        Ok(r) => r,
+        Err(_) => return None,
+    };
+
+    let buffer: BufReader<File> = BufReader::new(file);
+    let target_key: String = format!("<key>{name}</key>");
+    let mut in_package: bool = false;
+    let mut dict_level: u8 = 0;
+    let mut parse_str: bool = false;
+    for line in buffer.lines() {
+        if line.is_err() {
+            continue;
+        }
+        let line = line.unwrap();
+        if line.is_empty() {
+            continue;
+        }
+        let line = line.trim();
+
+        if line == "<dict>" && in_package { 
+            dict_level += 1;
+        }
+        if line == "</dict>" && in_package { 
+            dict_level -= 1;
+            if dict_level == 0 {
+                continue;
+            }
+        }
+        if line == target_key {
+            in_package = true;
+            dict_level += 1;
+            continue;
+        }
+        
+        if in_package {
+            if line == "<key>pkgver</key>" {
+                // Next line is the version
+                parse_str = true;
+                continue;
+            }
+            if parse_str {
+                let string: String = line[8..line.len() - 9].to_string();
+                // {package-name}-{ver}_{rev} 
+                let no_rev: &str = string.split('_').next().unwrap();
+                let version: &str = no_rev.split('-').last().unwrap();
+
+                return Some(version.to_string());
+            }
+        }
+    }
+
+    None
+}
+
+fn substitite_package_name(name: &str) -> &str {
+    // Substitutes a executable name for the package's name
+    // E.g turns nvim to neovim 
+    match name {
+        "nvim" => "neovim",
+        _ => name
+    }
+}
+
+
+
+// Known Hashes
+// Please contribute these so I'm not mind-numbingly doing these
+// 
+// Oh, and to the nerds who want to critisize this detection method in issues; this is a last ditch
+// resort to running {program} --version, as running commands like that is likely to cause really
+// really fuckin bad performance, but if we already know the hashes, we don't even have to do that. 
+// I'm sorry if it upsets you that this isn't going to catch every package's binary with every
+// build paramater and difference.
+fn compare_hash(hash: &str) -> Option<String> {
+    match hash {
+        // Kitty
+        "bfc1a826895089928bd40eb09a340c6f3b6eb22d51589ca32c032761ff44843b" => Some("0.35.2".to_string()), // Arch
```

Comment:
You are f**king cheating with performance.

Reply:
> What makes CrabFetch different is that it aims to be as performant as possible while still remaining easy to use

Full of cheats

Reply:
I've added a `use_version_checksums` config option to master to toggle using them. Sorry if my attempt of thinking about the problem differently has caused any offense.